Tests/SchemaMigration/TestSchemaMigration.m

```diff
@@ -0,0 +1,1142 @@
+		UKIntsEqual(0, [[[[newChild entityDescription] parent] owner] version]);
+		
+		UKObjectsEqual(@"Test", newParent.storeItem.packageName);
+		UKObjectsEqual(@"Test", newParent.entityDescription.owner.name);
+		UKObjectsEqual(@"org.etoile-project.CoreObject", newParent.entityDescription.parent.owner.name);
+		UKObjectsEqual(@"Test", newChild.storeItem.packageName);
+		UKObjectsEqual(@"Test", newChild.entityDescription.owner.name);
+		UKObjectsEqual(@"org.etoile-project.CoreObject", newChild.entityDescription.parent.owner.name);
+	}];
+}
+
+- (COSchemaMigration *)registerMigrationWithVersion: (int64_t)version
+                                             packageName: (NSString *)package
+					dependentSourceVersionsByPackageName: (NSDictionary *)deps
+                                              block: (COMigrationBlock)block
+{
+	COSchemaMigration *migration = [COSchemaMigration new];
+	
+	migration.packageName = package;
+	migration.destinationVersion = version;
+	migration.migrationBlock = block;
+	migration.dependentSourceVersionsByDomain = deps;
+	
+	[COSchemaMigration registerMigration: migration];
+	return migration;
+}
+
+- (COSchemaMigration *)registerMigrationWithVersion: (int64_t)version
+                                             package: (NSString *)package
+{
+	return [self registerMigrationWithVersion: version
+									   packageName: package
+			  dependentSourceVersionsByPackageName: @{}
+										block: NULL];
+}
+
+- (COSchemaMigration *)registerMigrationWithTestDomainVersion: (int64_t)version
+														block: (COMigrationBlock)block
+{
+	// The Test package depends on the CoreObject package, v0
+	return [self registerMigrationWithVersion: version
+									   packageName: @"Test"
+			  dependentSourceVersionsByPackageName: @{ @"org.etoile-project.CoreObject" : @(0) }
+										block: block];
+}
+
+- (void)testSchemaMigrationRegistration
+{
+	COSchemaMigration *migration1 = [self registerMigrationWithVersion: 500 package: @"Test"];
+	COSchemaMigration *migration2 = [self registerMigrationWithVersion: 501 package: @"Test"];
+	COSchemaMigration *migration3 =
+		[self registerMigrationWithVersion: 500 package: @"org.etoile-project.CoreObject"];
+
+	UKObjectsEqual(migration1, [COSchemaMigration migrationForPackageName: @"Test" destinationVersion: 500]);
+	UKObjectsEqual(migration2, [COSchemaMigration migrationForPackageName: @"Test" destinationVersion: 501]);
+	UKObjectsEqual(migration3, [COSchemaMigration migrationForPackageName: @"org.etoile-project.CoreObject"
+	                                              destinationVersion: 500]);
+	UKNil([COSchemaMigration migrationForPackageName: @"org.etoile-project.CoreObject"
+							 destinationVersion: 501]);
+}
+
+- (void)testInsertOrUpdateItemsWithoutMigration
+{
+	COObjectGraphContext *context = parent.objectGraphContext;
+	
+	UKDoesNotRaiseException([context insertOrUpdateItems: [context items]]);
+}
+
+/*	- (void) testInsertOrUpdateItemsWithMigration
```

Comment:
Basically just testing we get no exception when we are inserting or updating items that require a migration, and the resulting inner objects are identical to the one we would get by loading the items from the store, as we do in the other test methods.